### PR TITLE
enhance: resource group declarative api

### PR DESCRIPTION
**Pull request by chyezh**
Related Issue: #244
Related Milvus Issue: https://github.com/milvus-io/milvus/issues/30647

- Add resource group configuration proto.

- Add UpdateResourceGroups and modify AddResourceGroup api.

- Add NodeInfo.

### CODE REVIEW

File: proto/common.proto [hunk 1]
```diff
@@ -449,3 +452,10 @@ message ServerInfo {
     // reserved for newly-added feature if necessary.
     map<string, string> reserved = 6;
 }
+
+// NodeInfo is used to describe the node information.
+message NodeInfo {
+    int64 nodeID = 1;
+    string address = 2;
```
Comment (czs007) [inline proto/common.proto:459]: what is the difference between address and hostname?

**Comment by zhengbuqian** [comment]
this seems to be causing some golang tests to fail. in my local go unit test I built milvus using the latest milvus-proto commit on the master branch, and [`proxy_test.go`](https://github.com/milvus-io/milvus/blob/master/internal/proxy/proxy_test.go) complains `proxyTestServer` doesn't have method `UpdateResourceGroups`

**Pull request merged**